src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -283,7 +283,10 @@ public ICredentials Credentials
             {
                 return _credentials;
```

Comment:
We've had some requests that we parse the username/password out of URIs that might be set into these environment variables similar to how libcurl works (for explicit credentials anyways).

Example:

```
set HTTP_PROXY=http://username:password@mylocalproxy.corp.contoso.com:8080/
```

Reply:
Aren't we already doing that?
https://github.com/dotnet/corefx/blob/516f8508791f8648359841d5cc6ed9857e27c375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs#L109

Reply:
Thanks for pointing that out. So, it's just the Mac we aren't reading explicit credentials from system settings.

Do you recall if we have any unit tests for this? I.e. setting environment variables and then testing that the credentials are being properly set/get into the proxy's GetCredentials() method?

Reply:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/tests/UnitTests/HttpEnvironmentProxyTest.cs

?

Reply:
ok. thanks. Looks like we have unit tests but no end-to-end tests.